rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,109 @@
+# OPSGenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with OpsGenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in OpsGenie from access requests.
+Alerts were chosen over incidents in OpsGenie for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured using a toml file containing the required to interact with both Teleport access and the OpsGenie API.
+
+```
+[teleport]
+addr = "example.com:3025" # Teleport Auth Server GRPC API address
+client_key = "/var/lib/teleport/plugins/opsgenie/auth.key" # Teleport GRPC client secret key
+client_crt = "/var/lib/teleport/plugins/opsgenie/auth.crt" # Teleport GRPC client certificate
+root_cas = "/var/lib/teleport/plugins/opsgenie/auth.cas" # Teleport cluster CA certs
+
+[opsgenie]
+api_key = "key" # Opsgenie API Key
+
+[log]
+output = "stderr" # Logger output. Could be "stdout", "stderr" or "/var/lib/teleport/opsgenie.log"
+severity = "INFO" # Logger severity. Could be "INFO", "ERROR", "DEBUG" or "WARN".
+```
+
+### Getting an OpsGenie API key
+
+In the OpsGenie web UI go to Settings -> App settings -> API key management. Create key with Read, Create and Update access.
+
+### Executing
+
+The plugin will be started using a command of the form
+
+```
+Teleport opsgenie start –config <config file location>
+```
+
+## UX
+
+Once an access request has been created, the OpsGenie plugin will create an alert in service specified in the config file using the OpsGenie Alert API Create endpoint. 
```

Comment:
Does this mean that submitting an access request is going to page someone? Is that what we want?

Reply:
I've added an option to specify the severity to the config file so users can use that to determine whether or not the alerts will be created at a level that pages someone.